### PR TITLE
Add Obliterated75 Keyboard

### PR DESCRIPTION
**Pull request by awkannan**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cannonkeys/obliterated75/keymaps/via/rules.mk [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# rules.mk overrides to enable VIA
+
+VIA_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/rules.mk:4]:
```suggestion
VIA_ENABLE = yes
```

File: keyboards/cannonkeys/obliterated75/readme.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# CannonKeys Obliterated75
+
+Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)  
+Hardware Supported: STM32F072CBT6
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/readme.md:4]:
```suggestion
* Keyboard Maintainer: [Andrew Kannan](https://github.com/awkannan)
* Hardware Supported: STM32F072CBT6
```

File: keyboards/cannonkeys/obliterated75/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = STM32F072
+
+# Build Options
+#   comment out to disable the options.
+#
+
+
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/rules.mk:8]:
```suggestion
```

File: keyboards/cannonkeys/obliterated75/obliterated75.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_default( \
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/obliterated75.h:5]:
```suggestion
#define LAYOUT( \
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:34]:
```suggestion
	[_BASE] = LAYOUT(
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        KC_ESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_DEL, KC_INS,
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NUBS, KC_ENT, KC_PGUP,
+		KC_LSFT, KC_NUHS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_PGDN,
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, MO(_FN1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [_FN1] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:43]:
```suggestion
    [_FN1] = LAYOUT(
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        KC_ESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_GESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL, KC_TRNS, RGB_TOG,
+		KC_TRNS, KC_TRNS, KC_UP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_MOD,
+		BL_BRTG, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		BL_INC,  BL_DEC,  BL_TOGG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_GRV, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS, KC_TRNS
+    ),
+
+
+    [_FN2] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:53]:
```suggestion
    [_FN2] = LAYOUT(
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [_FN3] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:62]:
```suggestion
    [_FN3] = LAYOUT(
```

File: keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layer_names {
+    _BASE,
+    _FN1,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c:31]:
```suggestion
	[_BASE] = LAYOUT(
```

File: keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        KC_ESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_DEL, KC_INS,
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NUBS, KC_ENT, KC_PGUP,
+		KC_LSFT, KC_NUHS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_PGDN,
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, MO(_FN1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [_FN1] = LAYOUT_default(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c:40]:
```suggestion
    [_FN1] = LAYOUT(
```

File: keyboards/cannonkeys/obliterated75/info.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Obliterated75",
+    "url": "https://cannonkeys.com",
+    "maintainer": "awkannan",
+    "width": 16,
+    "height": 6.25,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/cannonkeys/obliterated75/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/cannonkeys/obliterated75/rules.mk [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
+CUSTOM_MATRIX = no # Custom matrix file
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+WS2812_DRIVER = spi
```
Comment (noroadsleft) [inline keyboards/cannonkeys/obliterated75/rules.mk:17]: Just alignment, no logic changes here.

```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
WS2812_DRIVER = spi
```

**Comment by awkannan** [comment]
OOF, I will change it to Layout_all later, when I am back at my PC

File: keyboards/cannonkeys/obliterated75/rules.mk [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+SLEEP_LED_ENABLE = yes      # Breathing sleep LED during USB suspend
```
Comment (zvecr) [inline keyboards/cannonkeys/obliterated75/rules.mk:12]:
```suggestion
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```

File: keyboards/cannonkeys/obliterated75/obliterated75.h [hunk 14]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/obliterated75.h:5]:
```suggestion
#define LAYOUT_all( \
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+    [_FN2] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [_FN3] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:62]:
```suggestion
    [_FN3] = LAYOUT_all(
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+        KC_ESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_GESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL, KC_TRNS, RGB_TOG,
+		KC_TRNS, KC_TRNS, KC_UP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RGB_MOD,
+		BL_BRTG, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		BL_INC,  BL_DEC,  BL_TOGG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_GRV, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET, KC_TRNS, KC_TRNS
+    ),
+
+
+    [_FN2] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:53]:
```suggestion
    [_FN2] = LAYOUT_all(
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+	[_BASE] = LAYOUT(
+        KC_ESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_DEL, KC_INS,
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NUBS, KC_ENT, KC_PGUP,
+		KC_LSFT, KC_NUHS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_PGDN,
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, MO(_FN1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [_FN1] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:43]:
```suggestion
    [_FN1] = LAYOUT_all(
```

File: keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+    _BASE,
+    _FN1,
+    _FN2,
+    _FN3
+};
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/via/keymap.c:34]:
```suggestion
	[_BASE] = LAYOUT_all(
```

File: keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+	[_BASE] = LAYOUT(
+        KC_ESC, KC_F1,   KC_F2, KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC, KC_DEL, KC_INS,
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
+		KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NUBS, KC_ENT, KC_PGUP,
+		KC_LSFT, KC_NUHS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_PGDN,
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, MO(_FN1), KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [_FN1] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c:40]:
```suggestion
    [_FN1] = LAYOUT_all(
```

File: keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum layer_names {
+    _BASE,
+    _FN1,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[_BASE] = LAYOUT(
```
Comment (fauxpark) [inline keyboards/cannonkeys/obliterated75/keymaps/default/keymap.c:31]:
```suggestion
	[_BASE] = LAYOUT_all(
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**